diff --git a/build.sh b/build.sh
index a23cf0a..b9a3029 100644
--- a/build.sh
+++ b/build.sh
@@ -240,7 +240,11 @@ if [ -n "${CC}" ] && [ -n "${CXX}" ]; then
    if [ "$os_type" = "Linux" ]; then
       final_binary="./python/interpret-core/interpret/lib/libebm_linux_x64.so"
    elif [ "$os_type" = "Darwin" ]; then
-      final_binary="./python/interpret-core/interpret/lib/libebm_mac_x64.dylib"
+      if [[ $target_platform == "osx-arm64" ]]; then
+         final_binary="./python/interpret-core/interpret/lib/libebm_mac_arm.dylib"
+      else
+         final_binary="./python/interpret-core/interpret/lib/libebm_mac_x64.dylib"
+      fi
    else
       printf "%s\n" "OS $os_type not recognized.  We support clang/clang++ on macOS and gcc/g++ on Linux"
       exit 1
